doc/sphinx-guides/source/developers/making-releases.rst

```diff
@@ -41,20 +47,39 @@ There are a variety of reasons why a step might deserve its own dedicated issue:
 
 Steps don't get their own dedicated issue if it would be confusing to have multiple people involved. Too many cooks in the kitchen, as they say. Also, some steps are so small the overhead of an issue isn't worth it.
 
-Before the release even begins you can coordinate with the project manager about the creation of these issues.
-
 .. |dedicated| raw:: html
 
       <span class="label label-success pull-left">
         Dedicated Issue
       </span>&nbsp;
 
+.. _declare-code-freeze:
+
 Declare a Code Freeze
 ---------------------
 
-The following steps are made more difficult if code is changing in the "develop" branch. Declare a code freeze until the release is out. Do not allow pull requests to be merged.
+When we declare a code freeze, we mean:
+
+- No additional features will be merged until the freeze is lifted.
+- Bug fixes will only be merged if they relate to the upcoming release in some way, such as fixes for regressions or performance problems in that release.
+- Pull requests that directly affect the release, such as bumping the version, will be merged, of course.
+
+The benefits of the code freeze are:
+
+- The team can focus on getting the release out together.
+- Regression and performance testing can happen on code that isn't changing.
+- The release notes can be written without having to worry about new features (and their release note snippets) being merged in.
+
+In short, the steps described below become easier under a code freeze.
 
-For a hotfix, a code freeze (no merging) is necessary not because we want code to stop changing in the branch being hotfix released, but because bumping the version used in Jenkins/Ansible means that API tests will fail in pull requests until the version is bumped in those pull requests.
+Note: for a hotfix, a code freeze is necessary not because we want code to stop changing in the branch being hotfix released, but because bumping the version used in Jenkins/Ansible means that API tests will fail in pull requests until the version is bumped in those pull requests. Basically, we want to get the hotfix merged quickly so we can propogate the version bump into all open pull requests so that API tests can start passing again in those pull requests.
+
+Push Back Milestones on Pull Requests That Missed the Train
+-----------------------------------------------------------
+
+As of this writing, we optimistically add milestones to issues and pull requests, hoping that the work will be complete before code freeze. Inevitably, we're a bit too optimistic.
+
+Hopefully, as the release approached, the team has already decided which pull requests (that aren't related to the release) won't make the cut. If not, go ahead and bump them to the next release.
 
 Conduct Performance Testing
```

Comment:
Should "Conduct Performance Testing" and "Conduct Regression Testing" be move down AFTER the new "deploy release candidate to QA" step? I think so but I wanted to check. My understanding is that all this testing is supposed to done on QA but I'm not sure.

Reply:
Right, make sense. I went ahead and moved the two testing steps (perf and regression) to go AFTER we deploy a release candidate to QA.